specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-04-01-preview/apimdeployment.json

```diff
@@ -475,6 +475,55 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/migrateToStv2": {
+      "put": {
+        "tags": [
+          "ApiManagementService"
+        ],
+        "operationId": "ApiManagementService_MigrateToStv2",
+        "description": "Upgrades an API Management service to the Stv2 platform. This change is not reversible. This is long running operation and could take several minutes to complete.",
+        "x-ms-examples": {
+          "ApiManagementMigrateService": {
+            "$ref": "./examples/ApiManagementServiceMigrateToStv2.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The service was successfully migrated.",
+            "schema": {
+              "$ref": "#/definitions/ApiManagementServiceResource"
+            }
+          },
+          "201": {
+            "description": "Migration request submitted.",
+            "schema": {
+              "$ref": "#/definitions/ApiManagementServiceResource"
```

Comment:
>ApiManagementServiceResource [](http://example.com/codeflow?start=37&length=28)

Why do you need to return the full resource in this case ?

Reply:
@VatsaPatel can you check if we are returning the full resource as response to 201. Or is it 202 for POST.

Reply:
It is full resource

Reply:
202 in itself doesnt have a resource body. The status needs to be reflected through the location header.

Reply:
use x-ms-long-running-operation-options to specify how the final state will be returned to the client

Reply:
This comment still needs to be addressed. Async response includes no payload. If there are multiple concurrent callers of this API, then subsequent calls should either fail, or return 200 with payload (with provisioningState set to a transition value).

Reply:
Removed the body from 202. Thanks!